src/main/java/Duke.java

```diff
@@ -1,10 +1,291 @@
+    }
+
+    public static void readData() throws FileNotFoundException {
+        File f = new File(DATA_FILE_PATH);
+        createFile();
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String line = s.nextLine();
+
+            if (line.startsWith("T")) {
+                String taskDisplay = line.substring(8);
+                stringList.add(listCount, taskDisplay);
+                Todo todoTask = new Todo(taskDisplay, 'T');
+                taskList.add(listCount, todoTask);
+                dueDate.add(listCount, null);
+
+                if (line.charAt(4) == '1') {
+                    taskList.get(listCount).markAsDone();
+                }
+            } else if (line.startsWith("D")) {
+                int deadlineIndex = line.indexOf("/");
+                String taskDisplay = line.substring(8, deadlineIndex);
+                stringList.add(listCount, taskDisplay);
+                String doBy = "(" + line.substring(deadlineIndex + 1) + ")";
+                Deadline deadlineTask = new Deadline(taskDisplay, 'D', doBy);
+                taskList.add(listCount, deadlineTask);
+                dueDate.add(listCount, line.substring(deadlineIndex + 1));
+
+                if (line.charAt(4) == '1') {
+                    taskList.get(listCount).markAsDone();
+                }
+            } else if (line.startsWith("E")) {
+                int eventIndex = line.indexOf("/");
+                String taskDisplay = line.substring(8, eventIndex);
+                stringList.add(listCount, taskDisplay);
+                String doBy = "(" + line.substring(eventIndex + 1) + ")";
+                Event eventTask = new Event(taskDisplay, 'E', doBy);
+                taskList.add(listCount, eventTask);
+                dueDate.add(listCount, line.substring(eventIndex + 1));
+
+                if (line.charAt(4) == '1') {
+                    taskList.get(listCount).markAsDone();
+                }
+            }
+            listCount++;
+        }
+    }
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    private static void appendToFile(String filePath, String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true); // create a FileWriter in append mode
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+    public static void writeData() {
+        try {
+            if (taskList.get(0).getTaskType() == 'T') {
+                String firstData = taskList.get(0).getTaskType() + " | " + taskList.get(0).getWrittenIcon() + " | " + stringList.get(0);
+                writeToFile(DATA_FILE_PATH, firstData + System.lineSeparator());
+            } else {
+                String firstData = taskList.get(0).getTaskType() + " | " + taskList.get(0).getWrittenIcon() + " | " + stringList.get(0) + " /" + dueDate.get(0);
```

Comment:
This line may be too long (>120 chars). Applies to any lines with >120 chars: https://se-education.org/guides/conventions/java/basic.html#layout